coalib/misc/Caching.py

```diff
@@ -0,0 +1,122 @@
+import calendar
+import time
+import hashlib
+
+from coalib.misc.CachingUtilities import (
+    pickle_load, pickle_dump, time_consistent, update_time_db,
+    delete_cache_files)
+
+
+class FileCache:
+
+    def __init__(self, log_printer, project_dir, flush_cache=False):
+        """
```

Comment:
`__init__` shouldnt have docstrings. The docstring for `__init__` should be added in the docstring of the class.
This is because when you do `help()` and stuff, the `__init__` function's docstring isn't shown.